plugins/experimental/stale_response/CacheUpdate.cc

```diff
@@ -0,0 +1,374 @@
+    TSMimeHdrFieldRemove(state->req_info->http_hdr_buf, state->req_info->http_hdr_loc, connection_hdr_loc);
+    TSMimeHdrFieldDestroy(state->req_info->http_hdr_buf, state->req_info->http_hdr_loc, connection_hdr_loc);
+    TSHandleMLocRelease(state->req_info->http_hdr_buf, state->req_info->http_hdr_loc, connection_hdr_loc);
+    connection_hdr_loc = connection_hdr_dup_loc;
+  }
+
+  TSDebug(PLUGIN_TAG, "[%s] {%u} Creating Connection:close hdr", __FUNCTION__, state->req_info->key_hash);
+  TSMimeHdrFieldCreateNamed(state->req_info->http_hdr_buf, state->req_info->http_hdr_loc, TS_MIME_FIELD_CONNECTION,
+                            TS_MIME_LEN_CONNECTION, &connection_hdr_loc);
+  TSMimeHdrFieldValueStringInsert(state->req_info->http_hdr_buf, state->req_info->http_hdr_loc, connection_hdr_loc, -1,
+                                  TS_HTTP_VALUE_CLOSE, TS_HTTP_LEN_CLOSE);
+  TSMimeHdrFieldAppend(state->req_info->http_hdr_buf, state->req_info->http_hdr_loc, connection_hdr_loc);
+  TSHandleMLocRelease(state->req_info->http_hdr_buf, state->req_info->http_hdr_loc, connection_hdr_loc);
+}
+
+/*-----------------------------------------------------------------------------------------------*/
+void
+get_pristine_url(StateInfo *state)
+{
+  TSHttpTxn txnp = state->txnp;
+  TSMBuffer hdr_url_buf;
+  TSMLoc url_loc;
+  // get the pristine url only works after remap state
+  if (TSHttpTxnPristineUrlGet(txnp, &hdr_url_buf, &url_loc) == TS_SUCCESS) {
+    int url_length;
+    char *url           = TSUrlStringGet(hdr_url_buf, url_loc, &url_length);
+    state->pristine_url = TSstrndup(url, url_length);
+    TSfree(url);
+    // release the buffer and loc
+    TSHandleMLocRelease(hdr_url_buf, TS_NULL_MLOC, url_loc);
+    TSDebug(PLUGIN_TAG, "[%s] {%u} pristine=[%s]", __FUNCTION__, state->req_info->key_hash, state->pristine_url);
+  } else {
+    TSDebug(PLUGIN_TAG_BAD, "[%s] {%u} TSHttpTxnPristineUrlGet failed!", __FUNCTION__, state->req_info->key_hash);
+  }
+}
+
+/*-----------------------------------------------------------------------------------------------*/
+bool
+intercept_get_key(TSMBuffer bufp, TSMLoc hdr_loc, const char *name, int name_len, string &key)
+{
+  bool retval      = false;
+  TSMLoc field_loc = TSMimeHdrFieldFind(bufp, hdr_loc, name, name_len);
+  if (field_loc) {
+    int value_len;
+    const char *value = TSMimeHdrFieldValueStringGet(bufp, hdr_loc, field_loc, 0, &value_len);
+    key.append(value, value_len);
+    retval = true;
+  }
+  TSHandleMLocRelease(bufp, hdr_loc, field_loc);
+  // TSDebug(PLUGIN_TAG, "[%s] key=[%s] found=%d",__FUNCTION__,name,key.c_str(),retval);
+  return retval;
+}
+
+/*-----------------------------------------------------------------------------------------------*/
+BodyData *
+intercept_check_request(StateInfo *state)
+{
+  uint32_t newKey      = 0;
+  BodyData *pBodyFound = nullptr;
+  TSHttpTxn txnp       = state->txnp;
+  uint32_t oldKey      = state->req_info->key_hash;
+
+#if (TS_VERSION_NUMBER >= 7000000)
```

Comment:
This is always true

Reply:
Good catch. I missed this when cleaning these out.